metricbeat/module/kafka/_meta/Dockerfile

```diff
@@ -1,9 +1,24 @@
-# TODO: No tags currently exist for this image. Tags should be used whever possible
-# as otherwise builds are not identical over time.
-FROM spotify/kafka
+FROM debian:stretch
 
-HEALTHCHECK --interval=1s --retries=90 CMD /opt/kafka_*/bin/kafka-topics.sh --zookeeper localhost:2181 --list
+ENV KAFKA_HOME /kafka
+ENV KAFKA_TOPIC=beats_topic_plain
+ENV KAFKA_LOGS_DIR="/kafka-logs"
+ENV KAFKA_VERSION 0.10.2.1
+ENV _JAVA_OPTIONS "-Djava.net.preferIPv4Stack=true"
+ENV TERM=linux
 
-EXPOSE 2181 9092
+RUN apt-get update && apt-get install -y curl openjdk-8-jre-headless netcat
 
-ENV ADVERTISED_HOST kafka
+RUN mkdir -p ${KAFKA_LOGS_DIR} && mkdir -p ${KAFKA_HOME} && curl -s -o $INSTALL_DIR/kafka.tgz \
+    "http://ftp.wayne.edu/apache/kafka/${KAFKA_VERSION}/kafka_2.11-${KAFKA_VERSION}.tgz" && \
+    tar xzf ${INSTALL_DIR}/kafka.tgz -C ${KAFKA_HOME} --strip-components 1
+
+ADD run.sh /run.sh
+
+EXPOSE 9092
+EXPOSE 2181
+
+#HEALTHCHECK --interval=1s --retries=90 CMD ${KAFKA_HOME}/bin/kafka-topics.sh --create --partitions 1 --replication-factor 1 --topic ${KAFKA_TOPIC} --zookeeper 127.0.0.1:2181
+HEALTHCHECK --interval=1s --retries=90 CMD ${KAFKA_HOME}/bin/kafka-topics.sh --zookeeper 127.0.0.1:2181 --list
```

Comment:
@ruflin this is the problem imo, it's not a valid `HEALTHCHECK`:

If I only have a running (and empty) Zookeeper, it still doesn't error, just tried it locally:

```sh
➜  kafka_2.12-0.11.0.0 bin/kafka-topics.sh --zookeeper=127.0.0.1:2181 --list 
[2017-11-14 11:35:07,659] INFO Accepted socket connection from /127.0.0.1:58444 (org.apache.zookeeper.server.NIOServerCnxnFactory)
[2017-11-14 11:35:07,664] INFO Client attempting to establish new session at /127.0.0.1:58444 (org.apache.zookeeper.server.ZooKeeperServer)
[2017-11-14 11:35:07,671] INFO Established session 0x15fba1822250002 with negotiated timeout 30000 for client /127.0.0.1:58444 (org.apache.zookeeper.server.ZooKeeperServer)
[2017-11-14 11:35:07,687] INFO Processed session termination for sessionid: 0x15fba1822250002 (org.apache.zookeeper.server.PrepRequestProcessor)
[2017-11-14 11:35:07,688] INFO Closed socket connection for client /127.0.0.1:58444 which had sessionid 0x15fba1822250002 (org.apache.zookeeper.server.NIOServerCnxn)
➜  kafka_2.12-0.11.0.0 echo $?                                               
0
```

Creating some random topic on the other hand is a valid check, it errors out when the broker isn't up:

```sh
➜  kafka_2.12-0.11.0.0 bin/kafka-topics.sh --zookeeper=127.0.0.1:2181 --create --partitions 1 --topic "foo" --replication-factor 1
[2017-11-14 11:39:20,379] INFO Accepted socket connection from /127.0.0.1:58468 (org.apache.zookeeper.server.NIOServerCnxnFactory)
[2017-11-14 11:39:20,384] INFO Client attempting to establish new session at /127.0.0.1:58468 (org.apache.zookeeper.server.ZooKeeperServer)
[2017-11-14 11:39:20,391] INFO Established session 0x15fba1822250004 with negotiated timeout 30000 for client /127.0.0.1:58468 (org.apache.zookeeper.server.ZooKeeperServer)
Error while executing topic command : replication factor: 1 larger than available brokers: 0
[2017-11-14 11:39:20,443] ERROR org.apache.kafka.common.errors.InvalidReplicationFactorException: replication factor: 1 larger than available brokers: 0
 (kafka.admin.TopicCommand$)
[2017-11-14 11:39:20,444] INFO Processed session termination for sessionid: 0x15fba1822250004 (org.apache.zookeeper.server.PrepRequestProcessor)
[2017-11-14 11:39:20,445] INFO Closed socket connection for client /127.0.0.1:58468 which had sessionid 0x15fba1822250004 (org.apache.zookeeper.server.NIOServerCnxn)
➜  kafka_2.12-0.11.0.0 echo $?
1
```

Reply:
Thanks for checking. Will try that out.